gen.py

```diff
@@ -136,47 +213,49 @@ class Country:
 }
 
 impl AsRef<str> for Country {
-  fn as_ref(&self) -> &str {
-    match self {
+    fn as_ref(&self) -> &str {
+        match self {
 {%- for country in countries %}
-      #[cfg(feature = "{{country.code}}")]
-      Country::{{country.code}} => "{{country.code}}",
+            #[cfg(feature = "{{country.code}}")]
+            Country::{{country|enum_name}} => "{{country|enum_name}}",
 {%- endfor %}
+        }
     }
-  }
 }
 
 impl std::str::FromStr for Country {
-  type Err = Error;
+    type Err = Error;
 
-  fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
-    Ok(match s {
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        Ok(match s {
 {%- for country in countries %}
-      #[cfg(feature = "{{country.code}}")]
-      "{{country.code}}" => Country::{{country.code}}, 
+            #[cfg(feature = "{{country.code}}")]
+            "{{country|enum_name}}" => Country::{{country|enum_name}},
 {%- endfor %}
-      _ => return Err(Error::CountryNotAvailable),
-    })
-  }
+            _ => return Err(Error::CountryNotAvailable),
+        })
+    }
 }
 
 """
 
 build = """
 use std::collections::HashSet;
 
-use crate::{data::*, prelude::*, HolidayMap, Result, Year};
+use crate::{data, prelude::*, HolidayMap, Result, Year};
 
 /// Generate holiday map for the specified countries and years.
 pub fn build(countries: Option<&HashSet<Country>>, years: Option<&std::ops::Range<Year>>) -> Result<HolidayMap> {
-  let mut map = HolidayMap::new();
+    let mut map = HolidayMap::new();
+
 {% for country in countries %}
-  #[cfg(feature = "{{country.code}}")]
-  if countries.is_none() || countries.unwrap().contains(&Country::{{country.code}}) {
-      map.insert(Country::{{country.code}}, {{country.code|escape}}::build(&years)?);
-  }
+    #[cfg(feature = "{{country.code}}")]
+    if countries.map_or(true, |c| c.contains(&Country::{{country|enum_name}})) {
```

Comment:
I find this quite hard to read :thinking:  If the `unwrap` is bothering you, use pattern matching instead?

```suggestion
    if matches!(countries, Some(c) if c.contains(&Country::{{country|enum_name}})) {
```

Reply:
Your suggestion doesn't work for the `None` case which should result in `true`. I created a seperate `should_build` function which is definitiv more readable